packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -409,100 +437,158 @@ function useMediaUploadQueue() {
           ITEM_STATUS.PENDING,
         ].includes(item.state)
     );
+
+    /**
+     * Whether any video transcoding is currently in progress.
+     *
+     * @type {boolean} Whether we're transcoding.
+     */
     const isTranscoding = state.queue.some(
       (item) => item.state === ITEM_STATUS.TRANSCODING
     );
+
+    /**
+     * Whether any video muting is currently in progress.
+     *
+     * @type {boolean} Whether we're muting.
+     */
     const isMuting = state.queue.some(
       (item) => item.state === ITEM_STATUS.MUTING
     );
+
+    /**
+     * Whether any video trimming is currently in progress.
+     *
+     * @type {boolean} Whether we're trimming.
+     */
     const isTrimming = state.queue.some(
       (item) => item.state === ITEM_STATUS.TRIMMING
     );
+
     /**
-     * Is a new resource being processed.
+     * Determine whether a resource is being processed.
      *
-     * @param {number} resourceId Resource id.
-     * @return {boolean} if resource with id is found.
+     * This is the case when an existing video in a story
+     * is being optimized/muted/trimmed, which will cause
+     * a new resource to be created and uploaded.
+     *
+     * As long as that resource's upload hasn't been cancelled,
+     * it is considered to be processing.
+     *
+     * @param {number} resourceId Resource ID.
+     * @return {boolean} Whether the resource is processing.
      */
     const isResourceProcessing = (resourceId) =>
       state.queue.some(
         (item) =>
-          [
-            ITEM_STATUS.PENDING,
-            ITEM_STATUS.UPLOADING,
-            ITEM_STATUS.TRANSCODING,
-            ITEM_STATUS.MUTING,
-            ITEM_STATUS.TRIMMING,
-          ].includes(item.state) && item.originalResourceId === resourceId
+          item.state !== ITEM_STATUS.CANCELLED &&
+          item.originalResourceId === resourceId
       );
+
     /**
-     * Is the current resource being processed.
+     * Determine whether the current resource is being processed.
      *
-     * @param {number} resourceId Resource id.
-     * @return {boolean} if resource with id is found.
+     * This is the case when uploading a new media item that first
+     * needs to be transcoded/muted/trimmed.
+     *
+     * As long as that resource's upload hasn't been cancelled,
+     * it is considered to be processing.
+     *
+     * @param {number} resourceId Resource ID.
+     * @return {boolean} Whether the resource is processing.
      */
     const isCurrentResourceProcessing = (resourceId) =>
       state.queue.some(
         (item) =>
-          [
-            ITEM_STATUS.PENDING,
-            ITEM_STATUS.UPLOADING,
-            ITEM_STATUS.TRANSCODING,
-            ITEM_STATUS.MUTING,
-            ITEM_STATUS.TRIMMING,
-          ].includes(item.state) && item.resource.id === resourceId
+          item.state !== ITEM_STATUS.CANCELLED &&
```

Comment:
Not called and not uploaded right?

Reply:
No, just checking for cancelled here because "uploaded" means the resource still needs to be updated everywhere.

It's only finished when it's either no longer in the queue or cancelled